src/main/webapp/search_area.js

```diff
@@ -135,13 +146,69 @@ class SearchArea {
     this.organizationObjectsList = await response.json();
   }
 
-  async setUrlParamValue(urlParamKey, urlParamValue) {
-    /* if the param is a zipcode, replace any existing one. Otherwise, add it to existing params */    
+  async setUrlParamValue(urlParamKey, urlParamValue) {   
+    /* New query value is not added if it is a duplicate or empty/null */
+    if (this.filterParams.getAll("filterParam").includes(urlParamValue) ||
+        this.filterParams.getAll("zipcode").includes(urlParamValue) ||
+        (urlParamValue === null) || (urlParamValue.trim() === "")) {
+      return;
+    }
+    
+    /* if the param is a zipcode, remove tag of any existing one & set new one*/ 
     if (urlParamKey === "zipcode") {
+      if (this.filterParams.get("zipcode")) {
+        /* If there is a zipcode being displayed, remove its tag so both aren't displayed */
+        this.removeFilterTag("zipcode", this.filterParams.get("zipcode"), document.getElementById("zipcodeTag"));
+      }
       this.filterParams.set(urlParamKey, urlParamValue);      
     } else {
       this.filterParams.append(urlParamKey, urlParamValue);      
     }
+    this.form.reset();
+    this.addFilterTag(urlParamKey, urlParamValue);
+    this.organizationObjectsList = [];
+    this.organizationListArea.innerHTML = "";
```

Comment:
.textContent

Reply:
This is to clear out the divs of organizationListArea